fpdcc/ccfp-asset-dashboard
# Improve the CIP + some minor changes

## Overview

Closes:
- #166 
- #167
- #170

Includes:
- changes in the CIP UI
    - added columns to the tables
    - added subrows to the tables to show the phase funding
        - some minor code tweaks to accommodate this
    - calculations for the portfolio summary
- backend:
    - calculations for the zone distribution via a new signal + tests
    - refactored the local asset signal (added a delete signal and refactored the code) 
    - added data to send in the portfolio view
- minor bug fixes discovered along the way (noted/explained in this PR's comments)
- Added an `actual_cost` field to the phase model based on feedback

## Testing Instructions
Apparently review apps don't work with heroku and github anymore? So download this branch and follow the steps in the readme to get it running locally. 

For testing:
- Create a phase or two.
    - Add some GIS assets.
    - Add some funding.
- Test the CIP planner. Make sure all of the calculations work, particularly the zone distributions.
    - Add/remove phases
    - Test the funding dropdown subrows
    - Test the name and saving of a portfolio
